lib/src/model.dart

```diff
@@ -31,22 +32,16 @@ class Model {
             // transform classes into model descriptions and loop through them
             modelDefinitions.forEach((m) {
                 // start entity
-                var entityName = CString(m["entity"]["name"]);
-                try {
-                    checkObx(bindings.obx_model_entity(_objectboxModel, entityName.ptr, m["entity"]["id"], m["entity"]["uid"]));
-                } finally {
-                    entityName.free();
-                }
+                var entityUtf8 = Utf8.toUtf8(m["entity"]["name"]);
+                var entityNamePointer = entityUtf8.cast<Uint8>();
+                checkObx(bindings.obx_model_entity(_objectboxModel, entityNamePointer, m["entity"]["id"], m["entity"]["uid"]));
```

Comment:
Note that Utf8.toUtf8 returns malloc-ed string so the resulting pointer must be freed after it has been passed to obx_model_entity()
```suggestion
                try {
                    checkObx(bindings.obx_model_entity(_objectboxModel, entityUtf8, m["entity"]["id"], m["entity"]["uid"]));
                } finally {
                    entityUtf8.free();
                }
```

Reply:
Hmm seeing all those try-finally-free I'm thinking if we could have a helper function like:

```dart
T withCString<T>(String string, T Function(Pointer<Utf8>) fn) {
    var utf8 = Utf8.toUtf8(string);
    try {
        return fn(utf8);
    } finally {
        utf8.free();
    }
}
```

Reply:
I assumed the ffi:Utf8 lib already freed the malloc.
The Function signatures are pretty specific, maybe if there's a general pattern, we can definitely use that helper function.

Reply:
I originally hoped the same but there seem to be no destructors in Dart so smart pointers are out of the question, unfortunately...